cds/webapi/C#/MetadataOperations/CDSWebApiServiceExtensions.cs

```diff
@@ -0,0 +1,2458 @@
+        /// <param name="componentId">ID of the solution component.</param>
+        /// <param name="componentType">The solution component to add to the unmanaged solution.</param>
+        /// <param name="solutionUniqueName">Unique name of the solution.</param>
+        /// <param name="addRequiredComponents">Indicates whether other solution components that are required by the solution component should also be added to the unmanaged solution.</param>
+        /// <param name="doNotIncludeSubComponents">Indicates whether the subcomponents should be included.</param>
+        /// <param name="includedComponentsSettingsValues">Any settings to be included with the component.</param>
+        /// <returns>The ID of the new solution component record.</returns>
+        public static Guid AddSolutionComponent(this CDSWebApiService svc,
+            Guid componentId,
+            SolutionComponentType componentType,
+            string solutionUniqueName,
+            bool addRequiredComponents,
+            bool? doNotIncludeSubComponents = null,
+            string includedComponentsSettingsValues = null)
+        {
+            Guid solutionComponentId;
+
+            if (!string.IsNullOrEmpty(solutionUniqueName))
+            {
+                //Add the attribute as part of a solution
+                JObject addSolutionComponentParameters = new JObject
+                {
+                    ["ComponentId"] = componentId,
+                    ["ComponentType"] = (int)componentType,
+                    ["SolutionUniqueName"] = solutionUniqueName,
+                    ["AddRequiredComponents"] = addRequiredComponents
+                };
+                if (doNotIncludeSubComponents != null)
+                {
+                    addSolutionComponentParameters["DoNotIncludeSubcomponents"] = doNotIncludeSubComponents;
+                }
+                if (includedComponentsSettingsValues != null)
+                {
+                    addSolutionComponentParameters["IncludedComponentSettingsValues"] = includedComponentsSettingsValues;
+                }
+
+                try
+                {
+                    JObject addSolutionComponentResponse = new JObject();
+
+                    addSolutionComponentResponse = svc.Post("AddSolutionComponent", addSolutionComponentParameters, STRONGCONSISTENCYHEADER);
+
+                    solutionComponentId = new Guid(addSolutionComponentResponse.GetValue("id").ToString());
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception($"There was a problem adding the {componentType} as a component to the {solutionUniqueName} solution:{ex.Message}");
+                }
+            }
+            else
+            {
+                throw new Exception("AddSolutionComponent requires the solutionUniqueName parameter value.");
+            }
+            return solutionComponentId;
+        }
+
```

Comment:
The next 3 methods can each be reduced to one call to svc.Post() with a inline anonymous JObject definition without the need to code a method. This is more a matter of the author's writing style and is just mentioned here as an FYI.

Reply:
Thanks. 
While each of these have a similar pattern, the goal is to show how to use the named actions rather than write a minimal amount of code.
[CanBeReferenced Action](https://docs.microsoft.com/en-us/dynamics365/customer-engagement/web-api/canbereferenced?view=dynamics-ce-odata-9)
[CanBeReferencing Action](https://docs.microsoft.com/en-us/dynamics365/customer-engagement/web-api/canbereferencing?view=dynamics-ce-odata-9)
[CanManyToMany Action](https://docs.microsoft.com/en-us/dynamics365/customer-engagement/web-api/canmanytomany?view=dynamics-ce-odata-9)